### DISCUSSION TITLE
Views not removing the old data

### DISCUSSION THREAD

**Discussion by k0rmo0** [apache/couchdb#4752] [Q&A] [answered]
I have a view that filters the data based on the date. In the beginning, it mapped the data correctly but, the next day it also had the data from the day before (the timestamp from the day before). Is there any way I could just have the data from the actual day and not from the day before? Here is the view map function:

```function (doc) {
  if (doc.timestamp && doc.license) {
    var today = new Date().setHours(0, 0, 0, 0);
    var thatDay = new Date(doc.timestamp * 1000).setHours(0, 0, 0, 0);
    if (today === thatDay) {
      //emit(doc.license, 1);
      emit("count", 1);
    }
  }
}

Reduce I use is count.

**Comment by rnewson** [comment]
a map function must depend solely on the input `doc` parameter. You've used the current time (of indexing) and this is why your results are wrong.

The map functions run over each document at index time, not query time, so "today" will be yesterdays date for docs indexed yesterday for queries made today.

**Reply by k0rmo0** [reply]
Okay, thank you. But is there any way I can get the data from the current date, or this is just the wrong approach?

**Comment by rnewson** [comment, marked as answer]
```
function(doc) {
    var docDay = new Date(doc.timestamp * 1000).setHours(0, 0, 0, 0);
    emit(docDay);
}
```

is all you need, you can then get the count of documents for any given day with `?reduce=true&key=1694127600000` for example.

You first build an index of your document contents and then query the range you want. 

In your original formulation CouchDB would need to reindex every document every time you queried, and we didn't design it that way for hopefully obvious reasons. :)

**Reply by k0rmo0** [reply]
Thank you :D
